src/history/hs_rec.c

```diff
@@ -519,36 +497,35 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_MULTI *mult
              */
             if (upd->type == WT_UPDATE_STANDARD && F_ISSET(upd, WT_UPDATE_HS))
                 break;
+
+            /*
+             * Save the first non timestamped update in the update chain. This is used to reset all
+             * the following update timestamps in the chain. Do not save if it is the last update in
+             * the chain. While inserting this last update, all the existing history store updates
+             * will reinsert after correcting their timestamps.
```

Comment:
all the existing history store updates will reinsert after correcting their timestamps

The wording is weird.

Reply:
Fixed.